### PR TITLE
[core] Check existence before expire: more check.

### PR DESCRIPTION
**Pull request by leaves12138**
### Purpose

Two place need to check

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java [hunk 1]
```diff
@@ -213,8 +214,25 @@ public int expireUntil(long earliestId, long endExclusiveId) {
...
+            // the end exclusive snapshot is gone
+            // there is no need to proceed
+            return 0;
+        }
+
+        Set<String> skippingSet = new HashSet<>();
+        try {
+            skippingSet.addAll(snapshotDeletion.manifestSkippingSet(skippingSnapshots));
+        } catch (Exception e) {
+            // mayby snapshot been deleted by other jobs.
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java:230]: mayby?

File: paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java [hunk 2]
```diff
@@ -213,8 +214,25 @@ public int expireUntil(long earliestId, long endExclusiveId) {
...
+            if (e.getCause() == null || !(e.getCause() instanceof FileNotFoundException)) {
+                throw e;
+            }
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java:233]: Add a log here?
Explain why we catch Exception here and no longer throw them out.

File: paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java (hunk 1 above)
Comment (leaves12138) [reply paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java:230]: See https://github.com/apache/paimon/pull/4235

**Review by yuzelin — APPROVED** [review]
+1

File: paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java (hunk 1 above)
Comment (wwj6591812) [reply paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java]: OK

**Pull request merged**